primary/src/primary.rs

```diff
@@ -65,6 +66,7 @@ pub enum PrimaryMessage<PublicKey: VerifyingKey> {
 
     PayloadAvailabilityResponse {
         payload_availability: Vec<(CertificateDigest, bool)>,
+        from: PublicKey,
```

Comment:
Unfortunately until we have authentication between the nodes we'll have to include the peer's name/key between the requests/responses for messages that need to identify per peer - which is of course unsafe.